crates/iroha_data_model/src/account.rs

```diff
@@ -77,6 +77,24 @@ mod model {
         pub metadata: Metadata,
     }
 
+    /// Read-only reference to [`Account`].
+    /// Used in query filters to avoid copying.
+    pub struct AccountEntry<'world> {
+        /// Identification of the [`Account`].
+        pub id: &'world AccountId,
+        /// Metadata of this account as a key-value store.
+        pub metadata: &'world Metadata,
+    }
```

Comment:
`*Entry` structs add quite a bit of duplication. Neither terrible nor great, but maybe it's fine..

I wonder what would be the impact of changing the original struct to hold copy-on-write pointers to the values instead (i.e. `Account` would hold `Cow<'_, AccountId>` and `Cow<'_, Metadata>`). It would allow creation of `Account` from references, and ability to avoid a loooot of `.clone`s around the codebase. However, I see that it would require to put lifetime mentions just everywhere.

Reply:
I think there will be a lot of problems with lifetimes
* I don't know simple way to use lifetimes in `type_descriptions!` macro
* Maps in `World` will be like `Map<AccountId, (Cow<'_, AccountId>, ...other fields)>` so map values should reference keys and I am not sure that it is possible